prow/github/hmac.go

```diff
@@ -46,3 +64,37 @@ func PayloadSignature(payload []byte, key []byte) string {
 	sum := mac.Sum(nil)
 	return "sha1=" + hex.EncodeToString(sum)
 }
+
+// For given repository/organization, returns all *valid* HMAC tokens. Validity is determined based on time till they are valid.
+func ExtractHmacs(repo string, tokenGenerator func() []byte) [][]byte {
+	t := tokenGenerator()
+	repoToTokenMap := map[string]map[string]time.Time{}
+
+	err := yaml.Unmarshal(t, &repoToTokenMap)
+	if err != nil {
+		fmt.Printf("error: %v", err)
```

Comment:
Logging library, and also be specific about what errored.

```suggestion
		logrus.WithError(err).Error("Couldn't unmarshal the hmac secret")
```